fdbserver/storageserver.actor.cpp

```diff
@@ -1100,29 +1112,31 @@ class StorageServer : public StorageServerBase {
 
 namespace ptxn {
 
+const KeyRef SUBSCRIBED_STORAGE_TEAMS_KEY = "\xff\xff/storageserver/lastSubscribedStorageTeams"_sr;
+
 class StorageServer : public StorageServerBase {
 public:
 	std::shared_ptr<ptxn::PeekCursorBase> logCursor;
 
+	// The key used to inform this storage server the new list of storage teams.
+	KeyRef storageServerToTeamIDKey;
+
+	// The storage teams the storage server has subscribed
 	StorageServerStorageTeams storageServerStorageTeams;
 
-	const std::unordered_set<StorageTeamID>& getSubscribedStorageTeamIDs() const;
+	// Gets the storage teams the storage server has subscribed, excluding the private mutation team.
+	const std::set<StorageTeamID>& getSubscribedStorageTeamIDs() const;
 
 	StorageServer(IKeyValueStore* pKVStore,
 	              const Reference<const AsyncVar<ServerDBInfo>>& serverDBInfo,
 	              const StorageServerInterface& storageServerInterface,
 	              const StorageTeamID& privateMutationsStorageTeamID_)
 	  : StorageServerBase(pKVStore, serverDBInfo, storageServerInterface),
 	    storageServerStorageTeams(privateMutationsStorageTeamID_) {}
-
-	void updateSubscribedStorageTeamIDs();
 };
 
-const std::unordered_set<StorageTeamID>& StorageServer::getSubscribedStorageTeamIDs() const {
-	ASSERT(logCursor);
-	auto ptr = dynamic_cast<merged::BroadcastedStorageTeamPeekCursorBase*>(logCursor.get());
-	ASSERT(ptr);
-	return ptr->getCursorStorageTeamIDs();
+const std::set<StorageTeamID>& StorageServer::getSubscribedStorageTeamIDs() const {
```

Comment:
This is from a previous request that the storage server should have an API that provides such information. Previously we grab it from the cursor, since we need to store such information, it is updated so it reads from local version of data.